### PR TITLE
Improve deployment-based create error handling

### LINKED ISSUE #2792: Deployment failure inner error not shown in many instances
Az CLI 2.0.2.

Run the following command:

```
az vm create -g JMP-AZ10-RHT --name bb-custom-vm100 --image rhel-custom-71 --admin-username azureuser --no-wait
```
Got this error:

>The template deployment 'vm_deploy_QcfOZ81VGBVOYQxXRfI9GjQJxtSBW8WY' is not valid according to the validation procedure. The tracking id is 'c82d598c-50c2-442b-8d0b-43c11da1479e'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details.

This error is non-actionable. The inner error, however, clearly indicates what failed in the validation:

```json
"statusMessage": "{\"error\":{\"code\":\"InvalidTemplateDeployment\",\"message\":\"The template deployment 'vm_deploy_QcfOZ81VGBVOYQxXRfI9GjQJxtSBW8WY' is not valid according to the validation procedure. The tracking id is 'c82d598c-50c2-442b-8d0b-43c11da1479e'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details.\",\"details\":[{\"code\":\"QuotaExceeded\",\"message\":\"Operation results in exceeding quota limits of Core. Maximum allowed: 100, Current in use: 100, Additional requested: 1.\"}]}}"
    },
```

Copy-pasting tail end of the above message: 

**Operation results in exceeding quota limits of Core. Maximum allowed: 100, Current in use: 100, Additional requested: 1.**

My request here is to pull this inner information and provide it in the CLI error message as it saves me loading the portal, going to activity log and finding it there. 

If there is already a way to get this in some detailed log of the az CLI execution, the error message should be expanded to provide the info on how to find it (e.g. azure CLI 1.0 was referring to azure.error.detailed file or something like that).

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #2792.

Ensures that the same exception handling logic is run when supplying `--no-wait` to template-based create commands: `vm`, `vmss`, `lb`, `application-gateway`, `vpn-connection`, `acs`, `availability-set`, `deployment`.

Adds table formatter to template-based creates that is compatible with the `--validate` flag (for both the success and error cases)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [N/A] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [N/A] Each command and parameter has a meaningful description.
- [N/A] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by tjprescott** [comment]
For an example, consider the following for VMSS create:
`az vmss create -g tjp-vm -n vmss1 --image UbuntuLTS --instance-count 1000`

**EXISTING BEHAVIOR**

*without --no-wait*
```
The template deployment 'vmss_deploy_vUrChnDcfKJ6TEtXTP8MbqKGGQqFoYqC' is not valid according to the validation procedure. The tracking id is '19e105b4-8712-4055-bbbf-3591c7a110e7'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details. Operation results in exceeding quota limits of Core. Maximum allowed: 100, Current in use: 20, Additional requested: 1000.
```

*with --no-wait*
```
The template deployment 'vmss_deploy_mUR9bwNH9cWKfxXfdjGGP3NtFV9I1i3O' is not valid according to the validation procedure. The tracking id is '5aca80ee-2abe-4805-9d3b-8c7dc81107c5'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details.
```

**NEW BEHAVIOR**

*with or without --no-wait*
```
Deployment failed. Tracking ID: ce415f53-f9a4-403e-9f2a-dd66097ecb5a. Operation results in exceeding quota limits of Core. Maximum allowed: 100, Current in use: 20, Additional requested: 1000.
```

*--validate success table format (previously returned nothing)*
```
Result     CorrelationId
---------  ------------------------------------
Succeeded  da4a8b1d-e1af-430f-bf60-b6b016a8e023
```

*--validate failure table format (previously returned nothing)*
```
Result                     TrackingId                            Message
-------------------------  ------------------------------------  --------------------------------------------------------------------------------------------------------------------------
InvalidTemplateDeployment  b9b73ae7-845e-44ae-8c80-fb3873f60915  Operation results in exceeding quota limits of Core. Maximum allowed: 100, Current in use: 21, Additional requested: 1000.
```

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 1]
```diff
@@ -125,6 +125,55 @@ def __setattr__(self, name, value):
...
+    except:  # pylint: disable=bare-except
+        error_message = '{} {}'.format(error_message, message)
+
+    cli_error = CLIError(error_message)
+    # capture response for downstream commands (webapp) to dig out more details
+    setattr(cli_error, 'response', getattr(ex, 'response', None))
+    raise cli_error
+
+
+def deployment_validate_table_format(result):
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: @derekbekoe can you recommend where these should go?  They are used by multiple command modules so I'd rather not duplicate them, but here seems like a pretty wonky fit.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -341,7 +344,7 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
     elif namespace.storage_profile == StorageProfile.SASpecializedOSDisk:
         required = ['os_type', 'attach_os_disk', 'use_unmanaged_disk']
         forbidden = ['os_disk_name', 'os_caching', 'data_caching', 'image', 'storage_account',
-                     'storage_container_name', 'data_disk_sizes_gb', 'storage_sku']
+                     'storage_container_name', 'data_disk_sizes_gb', 'storage_sku'] + auth_params
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: is this change related?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -1514,7 +1513,7 @@ def convert_av_set_to_managed_disk(resource_group_name, availability_set_name):
 
 # pylint: disable=too-many-locals, unused-argument, too-many-statements
 def create_vm(vm_name, resource_group_name, image=None, size='Standard_DS1_v2', location=None, tags=None, no_wait=False,
-              authentication_type=None, admin_password=None, admin_username=getpass.getuser(), ssh_dest_key_path=None,
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: This looks unrelated change. Out of curiosity though, why we remove the default?

File: test-object.json [hunk 4]
```diff
@@ -1,8 +0,0 @@
-[
-  {
```
Comment (yugangw-msft) [inline test-object.json]: unrelated change?

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 5]
```diff
@@ -125,6 +125,54 @@ def __setattr__(self, name, value):
...
+    if result.get('error', None):
+        error_result = OrderedDict()
+        error_result['result'] = result['error']['code']
+        tracking_id = re.match(r".*(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})", str(result['error']['message'])).group(1)
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: You might need to wrap it inside a try-catch block like you did for `handle_long_running_operation_exception`; otherwise the whole command might crash for unexpected error responses

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 2 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Hmm... no.  That's for a totally different branch.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py (hunk 3 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: This is for an unrelated change--weird.  But the reason we remove the default here is because it interferes with the validation.  I'll revert this.

File: test-object.json (hunk 4 above)
Comment (tjprescott) [inline test-object.json]: Kind of. These files snuck into an earlier PR and aren't necessary, which is why I'm removing them.

File: src/azure-cli-core/azure/cli/core/commands/__init__.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: It's okay. Would consider maybe moving it to `azure.cli.core.commands.util.py` as the code in `__init__.py` is more central to command loading etc.

Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: Since they are ARM-centric, I moved them into the commands.arm module.
